sources/packages/backend/apps/api/src/services/announcements/announcements.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from "@nestjs/common";
+import { Announcements, RecordDataModelService } from "@sims/sims-db";
+import { DataSource, LessThanOrEqual, MoreThanOrEqual } from "typeorm";
+
+@Injectable()
+export class AnnouncementsService extends RecordDataModelService<Announcements> {
+  constructor(dataSource: DataSource) {
+    super(dataSource.getRepository(Announcements));
+  }
+
+  /**
+   * Get system announcements.
+   * @returns system announcements list.
+   */
+  async getAnnouncements(): Promise<Announcements[]> {
+    const now = new Date();
+    return this.repo.find({
+      select: {
+        id: true,
+        messageTitle: true,
+        message: true,
+        target: true,
+        startDate: true,
+        endDate: true,
+      },
+      order: { startDate: "ASC" },
+      where: {
+        startDate: LessThanOrEqual(now.toDateString()),
```

Comment:
The toDateString() converts to user friendly date format which I doubt if typeorm can process accurately. 

![image](https://github.com/user-attachments/assets/23d9c9cb-05b2-48f5-9ac4-ba56b75e5ab5)

Reply:
Please use toISOString()

![image](https://github.com/user-attachments/assets/66b707d7-4123-4447-998f-368bd07064d6)

Reply:
removed